generator/src/lib.rs

```diff
@@ -90,11 +93,65 @@ pub fn derive_parser(input: TokenStream, include_grammar: bool) -> TokenStream {
         Err(error) => panic!("error parsing \n{}", error.renamed_rules(rename_meta_rule)),
     };
 
+    let grammar_docs = consume_grammar_doc(pairs.clone());
+    let line_docs = consume_line_docs(pairs.clone());
+
     let defaults = unwrap_or_report(validator::validate_pairs(pairs.clone()));
     let ast = unwrap_or_report(parser::consume_rules(pairs));
-    let optimized = optimizer::optimize(ast);
+    let optimized = optimizer::optimize(ast.clone());
+
+    let doc_comment = DocComment {
+        grammar_docs,
+        line_docs,
+        rules: ast,
+    };
+
+    generator::generate(
+        name,
+        &generics,
+        path,
+        optimized,
+        defaults,
+        &doc_comment,
+        include_grammar,
+    )
+}
+
+fn consume_grammar_doc(pairs: Pairs<'_, Rule>) -> Vec<&'_ str> {
+    let mut docs = vec![];
+    for pair in pairs {
+        if pair.as_rule() == Rule::grammar_doc {
+            docs.push(pair.as_str()[3..pair.as_str().len()].trim());
+        }
+    }
+
+    docs
+}
+
+fn consume_line_docs(pairs: Pairs<'_, Rule>) -> Vec<Vec<&'_ str>> {
+    let mut docs = vec![];
+    let mut comments = vec![];
+
+    for pair in pairs {
+        if pair.as_rule() == Rule::grammar_rule {
+            if let Some(inner) = pair.into_inner().next() {
+                if inner.as_rule() == Rule::line_doc {
+                    comments.push(inner.as_str()[3..inner.as_str().len()].trim());
```

Comment:
the magic number `3` is perhaps obvious in the context of this PR, but just for any potential future changes, it may be worth adding a small comment to note it corresponds to either `///` or `//!` (or perhaps extract it to a constant with a meaningful name, e.g. `DOC_PREFIX_LEN`)